carp.common/src/commonMain/kotlin/dk/cachet/carp/common/ddd/AggregateRoot.kt

```diff
@@ -9,6 +10,12 @@ import dk.cachet.carp.common.Immutable
  */
 abstract class AggregateRoot<TRoot, TSnapshot : Snapshot<TRoot>, TEvent : Immutable>
 {
+    /**
+     * The date when this object was created.
+     */
+    var creationDate: DateTime = DateTime.now()
```

Comment:
Isn't it possible to use `val` instead of `var` with `protected set`?

Reply:
I take my comment back. It is because `StudyRuntime` sets the date based on what comes from the `StudyRuntimeSnapshot`.